core/iwasm/common/wasm_runtime_common.c

```diff
@@ -2064,7 +2067,7 @@ wasm_runtime_init_wasi(WASMModuleInstanceCommon *module_inst,
                        const char *dir_list[], uint32 dir_count,
                        const char *map_dir_list[], uint32 map_dir_count,
                        const char *env[], uint32 env_count,
-                       char *argv[], uint32 argc,
+                       char *argv[], uint32 argc, int stdinfd, int stdoutfd, int stderrfd,
                        char *error_buf, uint32 error_buf_size)
 {
     uvwasi_t *uvwasi = NULL;
```

Comment:
uvwasi can also set fds, see [uvwasi_options_s](https://github.com/nodejs/uvwasi/blob/main/include/uvwasi.h#L64-L66) and [uvwasi_init](https://github.com/nodejs/uvwasi/blob/main/include/uvwasi.h#L71), had better set fds for uvwasi_options before calling uvwasi_init. For how to enable uvwasi, please ref to [config-libc in build_wamr](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/build_wamr.md#configure-libc).

Reply:
@Weining2019 
I tried to enable uvwasi, but i got some error
```
# cd <WAMR-ROOT>/core/deps
# git clone https://github.com/nodejs/uvwasi.git
# cd <WAMR-ROOT>/product-mini/platforms/linux-sgx/build
# cmake -DWAMR_BUILD_LIBC_UVWASI=1 -DWAMR_BUILD_PLATFORM=linux ..
# make
platforms/linux-sgx/build ‹main*› » make
[  2%] Built target vmlib_untrusted
[  5%] Building C object CMakeFiles/vmlib.dir/home/lifeng/WAMR/wasm-micro-runtime/core/deps/uvwasi/src/clocks.c.o
/home/lifeng/WAMR/wasm-micro-runtime/core/deps/uvwasi/src/clocks.c:3:11: fatal error: sys/time.h: No such file or directory
    3 | # include <sys/time.h>
      |           ^~~~~~~~~~~~
compilation terminated.
make[2]: *** [CMakeFiles/vmlib.dir/build.make:323: CMakeFiles/vmlib.dir/home/lifeng/WAMR/wasm-micro-runtime/core/deps/uvwasi/src/clocks.c.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:154: CMakeFiles/vmlib.dir/all] Error 2
make: *** [Makefile:130: all] Error 2
```

Reply:
Please "cd <WAMR-ROOT>/product-mini/platforms/linux" to build, but not linux-sgx.
To build-sgx, refer to [Build and Port WAMR vmcore (iwasm) for Linux SGX](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/linux_sgx.md).

Reply:
> Please "cd /product-mini/platforms/linux" to build, but not linux-sgx.
> To build-sgx, refer to [Build and Port WAMR vmcore (iwasm) for Linux SGX](https://github.com/bytecodealliance/wasm-micro-runtime/blob/main/doc/linux_sgx.md).

I build successfully under /product-mini/platforms/linux, Thanks